torch/fx/graph_module.py

```diff
@@ -179,6 +183,41 @@ def graph(self, val) -> None:
         self._graph = val
         self.recompile()
 
+    def to_folder(self, folder, module_name="FxModule"):
+        folder = Path(folder)
+        Path(folder).mkdir(exist_ok=True)
+        torch.save(self.state_dict(), folder / 'state_dict.pt')
+        tab = " " * 4
+        module_str = f"""
+import torch
+class {module_name}(torch.nn.Module):
+    def __init__(self):
+        super().__init__()
+        state_dict = torch.load('{folder}/state_dict.pt')
+"""
+        for module_name, module in self.named_children():
+            module_file = folder / f'{module_name}.pt'
+            torch.save(module, module_file)
+            nl = '\n'
+            module_str += f"{tab*2}self.{module_name} = torch.load('{module_file}') # {module.__repr__().replace(nl,' ')}\n"
+        for buffer_name, buffer in self._buffers.items():
+            module_str += f"{tab*2}self.{buffer_name} = state_dict['{buffer_name}']\n"
+
+        for param_name, param in self._parameters.items():
+            module_str += f"{tab*2}self.{param_name} = torch.nn.Parameter(state_dict['{param_name}'])\n"
```

Comment:
Can we emit a call to `self.load_state_dict(state_dict)` instead of manually populating Parameters/buffers?

Reply:
Don't `load_state_dict` assumes that your buffers/parameters already have the correct shapes and sizes, so it can just copy the data info.

I suppose we can just save it as a `torch.empty` of the correct shape - I'll do that.